client.go

```diff
@@ -410,19 +426,29 @@ func (c *Client) readLoop() error {
 
 // receiveData reads a message from session.
 func (c *Client) receiveData() ([]byte, error) {
+	type recv struct {
+		msg []byte
+		err error
+	}
+
 	session := c.getSession()
 	if session == nil {
 		return nil, errors.New("not connected")
 	}
 
-	msg, err := session.Recv()
-	if err != nil {
-		c.LocalKite.Log.Debug("Receive err: %s", err)
-	} else {
-		c.LocalKite.Log.Debug("Received : %s", msg)
-	}
+	done := make(chan recv)
```

Comment:
How about adding `stop` channel:
```go
	done := make(chan recv)

	go func() {
		msg, err := session.Recv()
		done <- recv{[]byte(msg), err}
	}()

	select {
	case r := <-done:
		return r.msg, r.err
	case err := <-c.interrupt:
		return nil, err
	}
```
vs
```go
	done, stop := make(chan recv), make(chan struct{})
        defer func() { close(stop) }()

	go func() {
		msg, err := session.Recv()
		select {
                case done <- recv{[]byte(msg), err}:
                case <-stop:
                }
	}()

	select {
	case r := <-done:
		return r.msg, r.err
	case err := <-c.interrupt:
		return nil, err
	}
```

I'm trying to avoid go-routine leak when session is interrupted.